### PR TITLE
Line generator for Charge-Bias Metrics

### PR DESCRIPTION
**Pull request by usert5432**
Hi @brettviren,

I am sending an RFC with the line track generator implementation for the new charge-bias metrics.
The purpose of this PR is to request feedback on:
1. Does this implementation/interface match the expectations we talked about?
2. Do people agree on the module/interface parameter names?
3. We discussed a possibility of loading wire-plane file and the need to write a module to handle them. Was this module written? So far, I request the user to explicitly provide the track midpoint coords.

Please let me know how this PR can be improved.

### CODE REVIEW

**Review by brettviren — APPROVED** [review]
The API looks fine to me.  I think it is good to merge when you are ready.

File: wirecell/gen/__main__.py
Comment (usert5432) [reply wirecell/gen/__main__.py:502]: Thank you. I will fix that!

File: wirecell/gen/linegen.py
Comment (usert5432) [reply wirecell/gen/linegen.py:14]: Yes, this are the comments for myself. I will remove them in the final version.

**Comment by brettviren** [comment]
@usert5432 life pre-empted this work for a while but I'm back to it and have updated wire-cell-python with what I hope will make it easier for you to finish the track generator.

I also added a new feature in wire-cell-data/detectors.jsonnet to provide a central registry of data files.  Doing a "git pull" on your clone of that will help.  In the case of wires, it lets the user specify just a canonical detector name, eg ~"pdsp"~ or a full wires file name.

This test hits on the main calls.

https://github.com/WireCell/wire-cell-python/blob/master/wirecell/util/test/test_wires_array.py#L16

I outline it:

- Load wires based a wires file name or a detector name if the registry is available.
- Get the wire end points as an `Nx2x3` array
- Correct the endpoints
- Get mean wire and pitch displacement over the array
- Construct the rotation matrix.
- Construct offset vector (may not be useful)

Then, in the last two plots I do some example operations using the rotation matrix and finding a point near the center of the central wire and offset by a fraction (0.5) of the pitch vector.  I guess specifying track center using this point will be most useful.  Default would be no offset from center of central wire.  Some study may look at fractional pitch offsets from that norm.

**Comment by usert5432** [comment]
Hi @brettviren,

I have implemented another interface `detlinegen` that takes a detector name, APA and WP indices and generates a track relative to the WP center. Could you please let me know if this interface conforms to the expectations and whether anything should be changed?

**Review by brettviren — APPROVED** [review]
All looks good.  

Once you are done, signify that by taking the PR out of "draft" status.

**Comment by usert5432** [comment]
Hi @brettviren, I have done some polishing to the line generator and I think it is ready for the final review. Please let me know if anything should be changed.

So far, I have kept the commit history to simplify review of changes. Please do not merge this version -- I will need properly squash the commits to make the history look neat.

**Review by brettviren — APPROVED** [review]
It looks fine.

We do not usually bother to squash.  You are of course free to do so.

**Comment by usert5432** [comment]
> It looks fine.

Thank you, @brettviren. I have squashed the commits and the PR is ready to be merged.

**Pull request merged**
